js/style/style.js

```diff
@@ -68,10 +76,32 @@ function Style(stylesheet, animationLoop) {
         this.fire('load');
     }.bind(this);
 
-    if (typeof stylesheet === 'string') {
-        ajax.getJSON(normalizeURL(stylesheet), loaded);
+    var sourceTypesLoaded = function(err) {
```

Comment:
Would [this comment](https://github.com/mapbox/mapbox-gl-js/pull/2667/files/3d6cf0a5ce5d118c52d2fe4446daffad25dd79b2#r65789289) tie in here?